semantic-conventions/src/tests/data/markdown/metrics_tables/expected.md

```diff
@@ -0,0 +1,16 @@
+# Test Markdown
+
+**`foo.size`**
+<!-- semconv metric.foo.size(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `foo.size` | Histogram | `{bars}` | Measures the size of foo. |
+<!-- endsemconv -->
```

Comment:
Can you please add a second metric to demonstrate how the table would be rendered here?
It would be one single table defining name+instrument+unit+description for all metrics and then below one table _each_ for the respective attributes, right?

Reply:
> Can you please add a second metric to demonstrate how the table would be rendered here? 

Sure, I agree this would make it clearer. I will add it in. 

> It would be one single table defining name+instrument+unit+description for all metrics and then below one table each for the respective attributes, right?

Take a look at the discussion around this comment: https://github.com/open-telemetry/build-tools/pull/79#issuecomment-1193448303

We decided that: 
* A "metric table" only has one metric. This means that each metric has it's own single-row table describing it's name, unit, description, and instrument. 
* Below a metric table will be an attribute table, associated with the above metric.

Reply:
I've added another metric and attribute 👍

Reply:
I think we could also define all metrics in one metrics table at once, and then have the respective attributes in their own table like this. This would still allow individual requirement levels:

```
## foo metrics
<!-- semconv metric.foo(metric_table) -->
| Name     | Instrument Type | Unit (UCUM) | Description    |
| -------- | --------------- | ----------- | -------------- |
| `foo.size` | Histogram | `{bars}` | Measures the size of foo. |
| `foo.duration` | Histogram | `{bazs}` | Measures the duration of foo. |
<!-- endsemconv -->

### Attributes for `metric.foo.size`
<!-- semconv metric.foo.size -->
| Attribute  | Type | Description  | Examples  | Requirement Level |
|---|---|---|---|---|
| `http.method` | string | HTTP request method. | `GET`; `POST`; `HEAD` | Required |
| `http.status_code` | int | [HTTP response status code](https://tools.ietf.org/html/rfc7231#section-6). | `200` | Conditionally Required: if and only if one was received/sent. |
<!-- endsemconv -->

### Attributes for `metric.foo.duration`
<!-- semconv metric.foo.duration -->
| Attribute  | Type | Description  | Examples  | Requirement Level |
|---|---|---|---|---|
| `http.method` | string | HTTP request method. | `GET`; `POST`; `HEAD` | Optional |
<!-- endsemconv -->
```

resulting in

---

## foo metrics

| Name     | Instrument Type | Unit (UCUM) | Description    |
| -------- | --------------- | ----------- | -------------- |
| `foo.size` | Histogram | `{bars}` | Measures the size of foo. |
| `foo.duration` | Histogram | `{bazs}` | Measures the duration of foo. |

### Attributes for `metric.foo.size`

| Attribute  | Type | Description  | Examples  | Requirement Level |
|---|---|---|---|---|
| `http.method` | string | HTTP request method. | `GET`; `POST`; `HEAD` | Required |
| `http.status_code` | int | [HTTP response status code](https://tools.ietf.org/html/rfc7231#section-6). | `200` | Conditionally Required: if and only if one was received/sent. |

### Attributes for `metric.foo.duration`

| Attribute  | Type | Description  | Examples  | Requirement Level |
|---|---|---|---|---|
| `http.method` | string | HTTP request method. | `GET`; `POST`; `HEAD` | Optional |

---

This can still be discussed and (if agreed on doing so) improved later on. Let's get this PR here in first 🙂 
Unless I miss something this would not break the YAML definitions but would only affect the generated markdown, which we can easily adapt in future iterations.

Reply:
I like this idea, and will help make it more concise. I think we should definitely do this, but also agree this should be done in a later PR. You are right that we can do this with the current yaml structure.

Reply:
I'm strongly against single-row tables. Instead of making it a table with multiple columns and a single row, why don't you make it a bullet-point list like

> **What would be the column header**: What would be the cell content below that header.

Alternatively, I think it would make more sense to have the `metric_table` be an argument to the metric_group's rendering, and then it would list all the metrics in that group. This is probably to be considered together with my comment about the `extends`-mechanism not really being ideal for declaring the association between metric and metric_group.

Reply:
If single-row tables is not desired, I think @arminru suggestion's on having the metrics together in a single table and splitting the attributes makes more sense. Not sure if I follow the bullet-point approach though, some examples would be nice to compare both approaches :)

Reply:
We all agree that single row tables aren't the way to go. We just don't want to extend the scope of this PR more. In a later PR, we can extend so it works for arbitrary amount of rows.

Reply:
Maybe we should just them agree on what will be the follow-up solution? Either the ones suggested by @arminru  or by @Oberon00, which I'm not sure I follow exactly.

Reply:
My second suggested solution was that you reference the metric group in the markdown, and it then renders a single table with multiple rows, one for each metric instrument that uses that metric group.

Reply:
If I understand correctly, I think both @arminru and @Oberon00 suggestions are the same or very similar. The `metric.foo` group in @arminru's example is the metric group that @Oberon00 is talking about where multiple metrics are listed. 

I definitely agree we should group metrics into a table and have separate attribute tables for each metric. I also think that we agree that this can be done in a later PR. This PR is now very bloated and reaches it's 1 year birthday in 2 days time.

Reply:
I agree that we should expand cleanup here later.  At this point, we just need something that works so we can define metric semantic conventions and share definitions with traces.  The most important aspect of this PR is *not* how it generates tables, which we can fix over time, but how the YAML is defined and shared between signals.  Let's focus on that as the most important part of this PR.




Also, regarding table generation, there are currently two main styles that exist *in our specification*:

1.  A table of metric names with a (then shared) table of attribute names that apply.
2.  A table of the metric where "empty" rows are created while attribtues are expanded (I didn't see this proposed).

I think #2 joined with what @jamesmoessis has w/ one-metric table kind of makes sense as you can see the cardinality of a metric pretty easily then.

However, let's continue tweaking how metric tables can be generated in a *follow up PR*